### PR TITLE
1055 Rebuilding supervisor dropdown and associated logic

### PR DESCRIPTION
**Pull request by robbdavis**
1055 Grad degree supervisor select - update key-value and template logic.

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php [hunk 1]
```diff
@@ -240,6 +240,8 @@
   </div>
 </div>
 
+<?php // hide the supervisor section if no supervisor required
+  if ($content['field_fgsif_supervisor_options']['#items'][0]['value'] != 'No supervisor is required'):?>
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: ideally, we should replace spaces in key with "-" or "_", not sure it is too late now

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.features.field_base.inc [hunk 2]
```diff
@@ -1548,10 +1548,10 @@ function ucgy_fgs_field_default_field_bases() {
...
-        'No supervisor is required for this program' => 'No supervisor is required for this program',
-        'A supervisor is required for this program and will be assigned to you' => 'A supervisor is required for this program and will be assigned to you',
-        'A supervisor is required for this program but is not required prior to beginning the program' => 'A supervisor is required for this program but is not required prior to beginning the program',
+        'A supervisor is required, and must agree to oversee the student\'s research before the student applies for admission' => 'A supervisor is required, and must agree to oversee the student\'s research before the student applies for admission',
+        'No supervisor is required' => 'No supervisor is required',
+        'A supervisor is required; potential supervisors must be listed when applying to the program' => 'A supervisor is required; potential supervisors must be listed when applying to the program',
+        'A supervisor is required, but is not required prior to the start of the program' => 'A supervisor is required, but is not required prior to the start of the program',
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.features.field_base.inc]: same here, should replace spaces in key with "-" or "_"

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.install [hunk 3]
```diff
@@ -98,3 +98,20 @@ function ucgy_fgs_update_7003()
...
+{
+  $fields = array(
+    'field_fgsif_supervisor'
+  );
+  foreach ($fields as $field) {
+    if ($instance = field_info_instance('node', $field, 'fgs_graduate_degree')) {
+      field_delete_instance($instance);
+    }
+  }
+}
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.install]: why we deleting 'fgs_graduate_degree' field, don't see in requirement.

File: sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php (hunk 1 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/templates/node--fgs-graduate-degree.tpl.php]: This is checking for the value. The key doesn't matter. Correct me if I'm wrong.

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.install (hunk 3 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.install]: This is the bundle / node name. Just part of the parameters for field_info_instance().

**Comment by balrajb** [comment]
conflicts need to resolve

**Comment by balrajb** [comment]
conflicts been resolved and key/value issue also fixed, thanks @robbdavis

**Pull request merged**